sdk/src/settings.rs

```diff
@@ -437,6 +645,49 @@ pub(crate) fn get_settings_value<'de, T: serde::de::Deserialize<'de>>(
     })
 }
 
+#[allow(unused)]
+pub(crate) fn get_profile_settings_value<'de, T: serde::de::Deserialize<'de>>(
+    value_path: &str,
+) -> Result<T> {
+    SETTINGS.with_borrow(|current_settings| {
+        PROFILE.with_borrow(|profile| {
+            if let Some(profile) = profile {
+                let value =
+                    current_settings.get::<T>(&format!("profile.{}.{}", profile, value_path));
+                match value {
+                    Ok(value) => return Ok(value),
+                    Err(err) => return Err(err.into()),
+                    // Ignore if the value isn't found and next check the default profile.
+                    Err(config::ConfigError::NotFound(_)) => {}
+                }
+            }
+
+            match current_settings.get::<T>(&format!("profile.default.{}", value_path)) {
+                Ok(value) => Ok(value),
+                Err(config::ConfigError::NotFound(_)) => Err(Error::NotFound),
+                Err(err) => Err(err.into()),
+            }
+        })
+    })
+}
+
+/// Returns the signer specified in the "signer" field of the currently active settings profile.
+///
+/// If the signer settings aren't specified, this function will return [Error::UnspecifiedSignerSettings][crate::Error::UnspecifiedSignerSettings].
+pub fn get_profile_settings_signer() -> Result<Box<dyn Signer>> {
```

Comment:
maybe just call this signer_from_settings(). In rust we don't usually use get_ for getters. we could use create_signer_from_settings() if we want an action for this.

Reply:
I'm not crazy about having a bunch of isolated global functions, so maybe we should have a public Settings class that has static methods?